### PR TITLE
expose some prototype transforms utils

### PR DESCRIPTION
**Pull request by pmeier**
In order to make it easy to subclass the v2 transforms, we also need to expose some utilities for public usage. This PR adds a new module `torchvision.prototype.transforms.utils` for that. Everything else stays in `_utils` as before.

For completeness, we also discussed dynamically create this module. This can be achieved by adding the following to `__init__.py`

```py
def _load_utils():
    import sys
    import types

    from torchvision.prototype.transforms import _utils

    utils = types.ModuleType(f"{__package__}.utils")
    utils.__package__ = __package__
    for attr in ["has_any", "has_all"]:  # TODO: expand this to everything we want to expose
        utils.__dict__[attr] = getattr(_utils, attr)
    sys.modules[utils.__name__] = utils

    return utils


utils = _load_utils()

del _load_utils
```

Now, one can do

```pycon
>>> from torchvision.prototype.transforms import utils
>>> utils.has_any
<function has_any at 0x7f92516060e0>
>>> from torchvision.prototype.transforms.utils import has_any
>>> has_any
<function has_any at 0x7f92516060e0>
```

This would save us the split between the two modules, but has some issues as well:

1. This is a fairly complex solution to an otherwise simple problem
2. Due to the dynamic creation, imports such as `from torchvision.prototype.transforms.utils import has_any` work perfectly fine, but are not picked up by static analyzers like `mypy` or IDE's

cc @vfdev-5 @datumbox @bjuncek

### CODE REVIEW

File: torchvision/prototype/transforms/utils.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    if not sizes:
+        raise TypeError("No image, video, mask or bounding box was found in the sample")
+    elif len(sizes) > 1:
+        raise ValueError(f"Found multiple HxW dimensions in the sample: {sequence_to_str(sorted(sizes))}")
+    h, w = sizes.pop()
+    return h, w
+
+
+def _isinstance(obj: Any, types_or_checks: Tuple[Union[Type, Callable[[Any], bool]], ...]) -> bool:
```
[the remark this answers is not in the document]
Comment (datumbox) [reply torchvision/prototype/transforms/utils.py:48]: I had a look on the usage of `_isinstance` in our codebase and feel that it's useful enough to be a public utility. WTDY?

We obviously can't remove the underscore, so we need a new name. Perhaps `check_type()`?

**Comment by datumbox** [comment]
When we merge this, we should update #6721 and #6534 to avoid breaking their code again.

File: torchvision/prototype/transforms/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmeier) [reply torchvision/prototype/transforms/utils.py:48]: Yes, I would expose it as well. +1 for `check_type`. @vfdev-5 Do you have an opinion here?

**Review by datumbox — APPROVED** [review]
@pmeier LGTM, thanks!

@vfdev-5 We are having issues with the [flaky](https://github.com/pytorch/vision/actions/runs/3582279419/jobs/6026319767) bbox resize tests. Could you please prioritise the fix?

**Comment by pmeier** [comment]
> We are having issues with the [flaky](https://github.com/pytorch/vision/actions/runs/3582279419/jobs/6026319767) bbox resize tests. Could you please prioritise the fix?

See #6995.

**Pull request merged**